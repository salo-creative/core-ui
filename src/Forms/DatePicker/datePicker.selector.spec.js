import React from 'react';
import 'jest-styled-components';

import { renderWithTheme } from '../../../test';

// COMPONENT
import DatePickerSelector from './datePicker.selector';

test('Renders against snapshot', async () => {
  const { container } = renderWithTheme(
    <DatePickerSelector />
  );
  expect(container.firstChild).toMatchSnapshot();
});