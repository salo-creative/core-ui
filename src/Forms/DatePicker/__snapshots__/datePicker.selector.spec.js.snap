// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders against snapshot 1`] = `
.c3 {
  position: relative;
  display: inline-block;
  vertical-align: middle;
  margin: 0;
  width: 100%;
  background: #fafcff;
}

.c2 {
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
}

.c4 {
  width: 100%;
  height: 4rem;
  background: transparent;
  color: #333333;
  font-size: inherit;
  padding: 0 4rem 0 1rem;
  border: 1px solid;
  border-color: rgba(0,172,237,0.5);
  margin: 0;
  border-radius: 0;
  text-overflow: '';
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none !important;
  border-radius: 0.4rem;
  font-size: 1.4rem;
  -webkit-transition: border 0.2s linear;
  transition: border 0.2s linear;
  position: relative;
  z-index: 2;
}

.c4:focus,
.c4:active {
  border-color: #00ACED;
}

.c4[disabled] {
  cursor: not-allowed;
  opacity: 0.75;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 4rem;
  height: calc(100% - 1.5rem);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  right: 0;
  top: 0.75rem;
  position: absolute;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-left: 1px solid;
  border-color: rgba(0,172,237,0.5);
  z-index: 1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: calc(100% + 2rem);
  margin: 0 -1rem;
  max-width: 40rem;
}

.c1 {
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 1rem;
  width: 33.3333%;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2 salo-select "
    >
      <div
        class="c3 salo-select__wrapper"
      >
        <select
          aria-invalid="false"
          aria-required="false"
          class="c4 salo-select__field"
          height="4rem"
          id="dd_undefined"
          name="dd_undefined"
        >
          <option
            value=""
          >
            DD
          </option>
          <option
            value="01"
          >
            01
          </option>
          <option
            value="02"
          >
            02
          </option>
          <option
            value="03"
          >
            03
          </option>
          <option
            value="04"
          >
            04
          </option>
          <option
            value="05"
          >
            05
          </option>
          <option
            value="06"
          >
            06
          </option>
          <option
            value="07"
          >
            07
          </option>
          <option
            value="08"
          >
            08
          </option>
          <option
            value="09"
          >
            09
          </option>
          <option
            value="10"
          >
            10
          </option>
          <option
            value="11"
          >
            11
          </option>
          <option
            value="12"
          >
            12
          </option>
          <option
            value="13"
          >
            13
          </option>
          <option
            value="14"
          >
            14
          </option>
          <option
            value="15"
          >
            15
          </option>
          <option
            value="16"
          >
            16
          </option>
          <option
            value="17"
          >
            17
          </option>
          <option
            value="18"
          >
            18
          </option>
          <option
            value="19"
          >
            19
          </option>
          <option
            value="20"
          >
            20
          </option>
          <option
            value="21"
          >
            21
          </option>
          <option
            value="22"
          >
            22
          </option>
          <option
            value="23"
          >
            23
          </option>
          <option
            value="24"
          >
            24
          </option>
          <option
            value="25"
          >
            25
          </option>
          <option
            value="26"
          >
            26
          </option>
          <option
            value="27"
          >
            27
          </option>
          <option
            value="28"
          >
            28
          </option>
          <option
            value="29"
          >
            29
          </option>
          <option
            value="30"
          >
            30
          </option>
          <option
            value="31"
          >
            31
          </option>
        </select>
        <span
          class="c5"
        >
          <svg
            style="width: 24px; height: 24px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M7,10L12,15L17,10H7Z"
              fill="#6c757d"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2 salo-select "
    >
      <div
        class="c3 salo-select__wrapper"
      >
        <select
          aria-invalid="false"
          aria-required="false"
          class="c4 salo-select__field"
          height="4rem"
          id="mm_undefined"
          name="mm_undefined"
        >
          <option
            value=""
          >
            MM
          </option>
          <option
            value="01"
          >
            Jan
          </option>
          <option
            value="02"
          >
            Feb
          </option>
          <option
            value="03"
          >
            Mar
          </option>
          <option
            value="04"
          >
            Apr
          </option>
          <option
            value="05"
          >
            May
          </option>
          <option
            value="06"
          >
            Jun
          </option>
          <option
            value="07"
          >
            Jul
          </option>
          <option
            value="08"
          >
            Aug
          </option>
          <option
            value="09"
          >
            Sep
          </option>
          <option
            value="10"
          >
            Oct
          </option>
          <option
            value="11"
          >
            Nov
          </option>
          <option
            value="12"
          >
            Dec
          </option>
        </select>
        <span
          class="c5"
        >
          <svg
            style="width: 24px; height: 24px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M7,10L12,15L17,10H7Z"
              fill="#6c757d"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2 salo-select "
    >
      <div
        class="c3 salo-select__wrapper"
      >
        <select
          aria-invalid="false"
          aria-required="false"
          class="c4 salo-select__field"
          height="4rem"
          id="yyyy_undefined"
          name="yyyy_undefined"
        >
          <option
            value=""
          >
            YYYY
          </option>
          <option
            value="2020"
          >
            2020
          </option>
          <option
            value="2019"
          >
            2019
          </option>
          <option
            value="2018"
          >
            2018
          </option>
          <option
            value="2017"
          >
            2017
          </option>
          <option
            value="2016"
          >
            2016
          </option>
          <option
            value="2015"
          >
            2015
          </option>
          <option
            value="2014"
          >
            2014
          </option>
          <option
            value="2013"
          >
            2013
          </option>
          <option
            value="2012"
          >
            2012
          </option>
          <option
            value="2011"
          >
            2011
          </option>
          <option
            value="2010"
          >
            2010
          </option>
          <option
            value="2009"
          >
            2009
          </option>
          <option
            value="2008"
          >
            2008
          </option>
          <option
            value="2007"
          >
            2007
          </option>
          <option
            value="2006"
          >
            2006
          </option>
          <option
            value="2005"
          >
            2005
          </option>
          <option
            value="2004"
          >
            2004
          </option>
          <option
            value="2003"
          >
            2003
          </option>
          <option
            value="2002"
          >
            2002
          </option>
          <option
            value="2001"
          >
            2001
          </option>
          <option
            value="2000"
          >
            2000
          </option>
          <option
            value="1999"
          >
            1999
          </option>
          <option
            value="1998"
          >
            1998
          </option>
          <option
            value="1997"
          >
            1997
          </option>
          <option
            value="1996"
          >
            1996
          </option>
          <option
            value="1995"
          >
            1995
          </option>
          <option
            value="1994"
          >
            1994
          </option>
          <option
            value="1993"
          >
            1993
          </option>
          <option
            value="1992"
          >
            1992
          </option>
          <option
            value="1991"
          >
            1991
          </option>
          <option
            value="1990"
          >
            1990
          </option>
          <option
            value="1989"
          >
            1989
          </option>
          <option
            value="1988"
          >
            1988
          </option>
          <option
            value="1987"
          >
            1987
          </option>
          <option
            value="1986"
          >
            1986
          </option>
          <option
            value="1985"
          >
            1985
          </option>
          <option
            value="1984"
          >
            1984
          </option>
          <option
            value="1983"
          >
            1983
          </option>
          <option
            value="1982"
          >
            1982
          </option>
          <option
            value="1981"
          >
            1981
          </option>
          <option
            value="1980"
          >
            1980
          </option>
          <option
            value="1979"
          >
            1979
          </option>
          <option
            value="1978"
          >
            1978
          </option>
          <option
            value="1977"
          >
            1977
          </option>
          <option
            value="1976"
          >
            1976
          </option>
          <option
            value="1975"
          >
            1975
          </option>
          <option
            value="1974"
          >
            1974
          </option>
          <option
            value="1973"
          >
            1973
          </option>
          <option
            value="1972"
          >
            1972
          </option>
          <option
            value="1971"
          >
            1971
          </option>
          <option
            value="1970"
          >
            1970
          </option>
          <option
            value="1969"
          >
            1969
          </option>
          <option
            value="1968"
          >
            1968
          </option>
          <option
            value="1967"
          >
            1967
          </option>
          <option
            value="1966"
          >
            1966
          </option>
          <option
            value="1965"
          >
            1965
          </option>
          <option
            value="1964"
          >
            1964
          </option>
          <option
            value="1963"
          >
            1963
          </option>
          <option
            value="1962"
          >
            1962
          </option>
          <option
            value="1961"
          >
            1961
          </option>
          <option
            value="1960"
          >
            1960
          </option>
          <option
            value="1959"
          >
            1959
          </option>
          <option
            value="1958"
          >
            1958
          </option>
          <option
            value="1957"
          >
            1957
          </option>
          <option
            value="1956"
          >
            1956
          </option>
          <option
            value="1955"
          >
            1955
          </option>
          <option
            value="1954"
          >
            1954
          </option>
          <option
            value="1953"
          >
            1953
          </option>
          <option
            value="1952"
          >
            1952
          </option>
          <option
            value="1951"
          >
            1951
          </option>
          <option
            value="1950"
          >
            1950
          </option>
          <option
            value="1949"
          >
            1949
          </option>
          <option
            value="1948"
          >
            1948
          </option>
          <option
            value="1947"
          >
            1947
          </option>
          <option
            value="1946"
          >
            1946
          </option>
          <option
            value="1945"
          >
            1945
          </option>
          <option
            value="1944"
          >
            1944
          </option>
          <option
            value="1943"
          >
            1943
          </option>
          <option
            value="1942"
          >
            1942
          </option>
          <option
            value="1941"
          >
            1941
          </option>
          <option
            value="1940"
          >
            1940
          </option>
          <option
            value="1939"
          >
            1939
          </option>
          <option
            value="1938"
          >
            1938
          </option>
          <option
            value="1937"
          >
            1937
          </option>
          <option
            value="1936"
          >
            1936
          </option>
          <option
            value="1935"
          >
            1935
          </option>
          <option
            value="1934"
          >
            1934
          </option>
          <option
            value="1933"
          >
            1933
          </option>
          <option
            value="1932"
          >
            1932
          </option>
          <option
            value="1931"
          >
            1931
          </option>
          <option
            value="1930"
          >
            1930
          </option>
          <option
            value="1929"
          >
            1929
          </option>
          <option
            value="1928"
          >
            1928
          </option>
          <option
            value="1927"
          >
            1927
          </option>
          <option
            value="1926"
          >
            1926
          </option>
          <option
            value="1925"
          >
            1925
          </option>
          <option
            value="1924"
          >
            1924
          </option>
          <option
            value="1923"
          >
            1923
          </option>
          <option
            value="1922"
          >
            1922
          </option>
          <option
            value="1921"
          >
            1921
          </option>
          <option
            value="1920"
          >
            1920
          </option>
          <option
            value="1919"
          >
            1919
          </option>
          <option
            value="1918"
          >
            1918
          </option>
          <option
            value="1917"
          >
            1917
          </option>
          <option
            value="1916"
          >
            1916
          </option>
          <option
            value="1915"
          >
            1915
          </option>
          <option
            value="1914"
          >
            1914
          </option>
          <option
            value="1913"
          >
            1913
          </option>
          <option
            value="1912"
          >
            1912
          </option>
          <option
            value="1911"
          >
            1911
          </option>
          <option
            value="1910"
          >
            1910
          </option>
          <option
            value="1909"
          >
            1909
          </option>
          <option
            value="1908"
          >
            1908
          </option>
          <option
            value="1907"
          >
            1907
          </option>
          <option
            value="1906"
          >
            1906
          </option>
          <option
            value="1905"
          >
            1905
          </option>
          <option
            value="1904"
          >
            1904
          </option>
          <option
            value="1903"
          >
            1903
          </option>
          <option
            value="1902"
          >
            1902
          </option>
          <option
            value="1901"
          >
            1901
          </option>
          <option
            value="1900"
          >
            1900
          </option>
        </select>
        <span
          class="c5"
        >
          <svg
            style="width: 24px; height: 24px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M7,10L12,15L17,10H7Z"
              fill="#6c757d"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
